### ISSUE TITLE
`event.params` is undefined in unit tests

### ISSUE THREAD

**Issue by zgosalvez** [firebase/firebase-functions#88]
### Version info
**firebase-functions:** 0.6.2
**firebase-tools:** 3.9.2
**firebase-admin:** 5.2.1

### Test case

```ts
//
// In unit test
//
// ...
event = {
  params: {
    userId: 'user-id',
  },
  data: new functions.database.DeltaSnapshot(null, admin, null, user)
};
console.log(event); // 1st log
await myFunctions.onUserWrite(event);
// ...
```
```ts
//
// In onUserWrite
//
export.onUserWrite = functions.database.ref('/users/userId').onWrite(async event => {
  // ...
  console.log(event); // 2nd log
  // ...
);
```

### Were you able to successfully deploy your functions?

No, since unit tests fail

### Expected behavior

1st log should equal 2nd log, showing both `event.params.userId` as `user-id`

### Actual behavior

1st log shows `event.params.userId` as `user-id`
2nd log shows `event.params.userId` as `undefined`; However, the `event.data` element is equal to the 1st log!

---

Reverting back to **firebase-functions:** 0.6.1 fixes the issue.

**Comment by Crazometer** [comment]
They changed the way the parameters are generated in https://github.com/firebase/firebase-functions/commit/3c67e24ccddf7996159111ec9cdeb226aae1bd2b. It now generates `params` from the snapshot ref. You can use this in your tests by doing

```js
new functions.database.DeltaSnapshot(null, admin, null, user, ref)
```

Not sure what the #154 in the linked commit comment refers to.

It would be nice if [this line](https://github.com/firebase/firebase-functions/commit/3c67e24ccddf7996159111ec9cdeb226aae1bd2b#diff-25e5fcb3f5ed7ffe6b1293fcab0149c6R86) could be changed  to not overwrite parameters. Our team is going to freeze the version for a while as we have a lot of tests that would need to be re-written.

**Comment by laurenzlong** [comment]
Hey @Crazometer and @zgosalvez , thanks for the report. This was an oversight on my part, I did not consider unit testing when I made the changes to params. I'm out of office for most of this week, so I'll fix it next week. You're also more than welcome to send a quick PR and I'll be happy to review.

**Comment by laurenzlong** [comment]
Fix will be in 0.6.3: https://github.com/firebase/firebase-functions/pull/93
